src/js/hooks/useUserHasPermissions.js

```diff
@@ -0,0 +1,75 @@
+import { useCallback, useMemo } from 'react';
+
+import _ from 'lodash';
+import { useSelector } from 'react-redux';
+
+import RoleType from 'consts/roleType';
+
+/**
+ * Returns true or false if current logged-in user has all the roles specified in the parameters.
+ * The function checks if user has either global or location based roles.
+ * @param minRequiredRole {RoleType} - minimum required role
+ * @param supplementalRoles {RoleType[]} [d=[]] - list of supplemental required roles
+ * @returns {boolean}
+ */
+const useUserHasPermissions = ({ minRequiredRole, supplementalRoles = [] }) => {
+  const {
+    currentUser,
+    currentLocationRoles,
+    isSuperuser,
+    isAdmin,
+    isApprover,
+    isRequestApprover,
+  } = useSelector((state) => ({
+    currentUser: state.session.user,
+    currentLocationRoles: state.session.currentLocationRoles,
+    isSuperuser: state.session.isSuperuser,
+    isAdmin: state.session.isUserAdmin,
+    isApprover: state.session.isUserApprover,
+    isRequestApprover: state.session.isUserRequestApprover,
+  }));
+
+  /**
+   * Returns true if user has all of the specified roles as global or location roles.
+   * @type {function(RoleType[]): boolean}
+   */
+  const hasEveryRole = useCallback((roles) => {
+    const userRoles = Array.isArray(currentUser?.roles) ? currentUser?.roles : [];
+    const userLocationRoles = Array.isArray(currentLocationRoles) ? currentLocationRoles : [];
+
+    const allUserRoles = new Set([...userRoles, ...userLocationRoles]);
+
+    return _.every(roles, (role) => allUserRoles.has(role));
+  }, [currentUser]);
```

Comment:
shouldn't the `currentLocation` also be a dependency?

Reply:
yes, true.
And currentUser dependancy should be `[currentUser.id]`

Reply:
exactly :D and the `currentLocation` -> `currentLocation?.id`